geopandas/tests/test_vectorized.py

```diff
@@ -0,0 +1,367 @@
+        from_shapely, serialize, deserialize, cysjoin)
+from shapely.geometry.base import (CAP_STYLE, JOIN_STYLE)
+
+import pytest
+import numpy as np
+
+
+triangles = [shapely.geometry.Polygon([(random.random(), random.random())
+                                       for i in range(3)])
+             for _ in range(10)]
+T = from_shapely(triangles)
+
+points = [shapely.geometry.Point(random.random(), random.random())
+          for _ in range(20)]
+P = from_shapely(points)
+
+
+point = points[0]
+
+
+def test_points():
+    x = np.arange(10).astype(np.float)
+    y = np.arange(10).astype(np.float) ** 2
+
+    points = points_from_xy(x, y)
+    assert (points.data != 0).all()
+
+    assert (x == points.x).all()
+    assert (y == points.y).all()
+
+    assert isinstance(points[0], shapely.geometry.Point)
+
+
+def test_from_shapely():
+    assert isinstance(T, GeometryArray)
+    assert [v.equals(t) for v, t in zip(T, triangles)]
+    # TODO: handle gc
+
+
+@pytest.mark.parametrize('attr,args', [
+    ('contains', ()),
+    ('covers', ()),
+    ('crosses', ()),
+    ('disjoint', ()),
+    ('equals', ()),
+    ('intersects', ()),
+    ('overlaps', ()),
+    ('touches', ()),
+    ('within', ()),
+    ('equals_exact', (0.1,))
+])
+def test_vector_scalar_predicates(attr, args):
+    triangles = [shapely.geometry.Polygon([(random.random(), random.random())
+                                           for i in range(3)])
+                 for _ in range(100)]
+
+    T = from_shapely(triangles)
+
+    point = shapely.geometry.Point(random.random(), random.random())
+    tri = triangles[0]
+
+    for other in [point, tri]:
+        result = getattr(T, attr)(other, *args)
+        assert isinstance(result, np.ndarray)
+        assert result.dtype == bool
+
+        expected = [getattr(tri, attr)(other, *args) for tri in triangles]
+
+        assert result.tolist() == expected
+
+
+@pytest.mark.parametrize('attr,args', [
+    ('contains', ()),
+    ('covers', ()),
+    ('crosses', ()),
+    ('disjoint', ()),
+    ('equals', ()),
+    ('intersects', ()),
+    ('overlaps', ()),
+    ('touches', ()),
+    ('within', ()),
+    ('equals_exact', (0.1,))
+])
+def test_vector_vector_predicates(attr, args):
+    A = [shapely.geometry.Polygon([(random.random(), random.random())
+                                   for i in range(3)])
+         for _ in range(100)]
+    B = [shapely.geometry.Polygon([(random.random(), random.random())
+                                   for i in range(3)])
+         for _ in range(100)]
+
+
+    vec_A = from_shapely(A)
+    vec_B = from_shapely(B)
+
+    result = getattr(vec_A, attr)(vec_B, *args)
+    assert isinstance(result, np.ndarray)
+    assert result.dtype == bool
+
+    expected = [getattr(a, attr)(b, *args) for a, b in zip(A, B)]
+
+    assert result.tolist() == expected
+
+
+@pytest.mark.parametrize('attr', [
+    'boundary',
+    'centroid',
+#    'representative_point',
```

Comment:
did this fail? or was it not yet implemented? 
in the meantime at least it seems included in vectorized.pyx